packages/ui/core/src/app/modules/interfaces/interfaces.component.ts

```diff
@@ -0,0 +1,250 @@
+        ),
+        tap((result: { type: 'markdown' | 'file'; value: unknown }) => {
+          if (result.type === 'markdown') {
+            this.markdownResponse = result.value as string;
+          } else if (result.type === 'file') {
+            // Your base64 string
+            const base64String = result.value as string;
+            // Splitting the base64 string to extract MIME type
+            const mimeType = base64String.match(
+              /data:([a-zA-Z0-9]+\/[a-zA-Z0-9-.+]+).*,.*/
+            );
+            let extension = '';
+            if (mimeType) {
+              extension = this.mimeToExtension(mimeType[1]);
+            }
+            const base64Data = base64String.split(',')[1];
+            const byteCharacters = atob(base64Data);
+            const byteNumbers = new Array(byteCharacters.length);
+            for (let i = 0; i < byteCharacters.length; i++) {
+              byteNumbers[i] = byteCharacters.charCodeAt(i);
+            }
+            const byteArray = new Uint8Array(byteNumbers);
+            const blob = new Blob([byteArray]);
+            const url = URL.createObjectURL(blob);
+            const link = document.createElement('a');
+            link.href = url;
+            link.download = `${this.title}.${extension}`; // Set your desired file name here
+            link.click();
+            // Clean up by revoking the object URL
+            URL.revokeObjectURL(url);
+          }
+          this.loading = false;
+        })
+      );
+    }
+  }
+
+  getInputKey(str: string) {
+    return str
+      .replace(/\s(.)/g, function ($1) {
+        return $1.toUpperCase();
+      })
+      .replace(/\s/g, '')
+      .replace(/^(.)/, function ($1) {
+        return $1.toLowerCase();
+      });
+  }
+
+  buildTextInputs(inputs: Input[]) {
+    inputs.forEach((prop) => {
+      this.textInputs.push(Property.ShortText(prop) as unknown as Input);
+      this.interfaceForm.addControl(
+        this.getInputKey(prop.displayName),
+        new FormControl('', {
+          nonNullable: prop.required,
+          validators: prop.required ? [Validators.required] : [],
+        })
+      );
+    });
+  }
+
+  buildFileInputs(inputs: Input[]) {
+    inputs.forEach((prop) => {
+      this.fileInputs.push(Property.File(prop) as unknown as Input);
+      this.interfaceForm.addControl(
+        this.getInputKey(prop.displayName),
+        new FormControl('', {
+          validators: prop.required ? [Validators.required] : [],
+        })
+      );
+    });
+  }
+
+  toBase64(file: File): Observable<string> {
+    return new Observable((observer: Observer<string>) => {
+      const reader = new FileReader();
+
+      reader.readAsDataURL(file);
+
+      reader.onload = () => {
+        observer.next(reader.result as string);
+        observer.complete();
+      };
+
+      reader.onerror = (error) => {
+        observer.error(error);
+      };
+    });
+  }
+
+  mimeToExtension(mimeType) {
```

Comment:
Use that  https://www.npmjs.com/package/mime-types, we already have it installed